src/Microsoft.ML.Data/Transforms/RangeFilter.cs

```diff
@@ -153,7 +153,7 @@ private RangeFilter(IHost host, ModelLoadContext ctx, IDataView input)
 
             _type = schema[_index].Type;
             if (_type != NumberDataViewType.Single && _type != NumberDataViewType.Double && _type.GetKeyCount() == 0)
-                throw Host.ExceptSchemaMismatch(nameof(schema), "source", column, "float, double or KeyType", _type.ToString());
+                throw Host.ExceptSchemaMismatch(nameof(schema), "source", column, "Single, Double or KeyType", _type.ToString());
```

Comment:
>KeyType [](start = 101, length = 7)

So, this is an interesting one... the *type* is of key, but we are calling it `KeyType`... yet, we say `Single` not `SingleType`. Probably not worth resolving now, but something to think about in future, no doubt.

Reply:
while `Single` and `Double` is `.net` types, Key is something we introduce. So I fill need to distinguish them somehow.
In same time Eric has PR to rename `KeyType` to `KeyDataViewType`

---
In reply to: [268028603](https://github.com/dotnet/machinelearning/pull/3046#discussion_r268028603) [](ancestors = 268028603)

Reply:
Ok, let's name it Key.

---
In reply to: [268253632](https://github.com/dotnet/machinelearning/pull/3046#discussion_r268253632) [](ancestors = 268253632,268028603)